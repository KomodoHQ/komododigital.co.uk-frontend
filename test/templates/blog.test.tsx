import React from 'react';
import Blog from '../../src/templates/blog';
import renderer from 'react-test-renderer';
import GatsbyData from '../../gatsby-config';

test('Blog template is rendered', () => {
  const hocProps = {
    imagesource: 'https://www.komododigital.co.uk/static/media/logo.50a4b652.svg',
    html: '<p>Test</p>',
    data: {
      site: GatsbyData,
      logo: {
        childImageSharp: {
          fixed: {
            height: 29,
            src: "",
            srcSet: "",
            srcSetWebp: "",
            srcWebp: "",
            tracedSVG: "",
            width: 170
          }
        }
      },
      logo_inverted: {
        childImageSharp: {
          fixed: {
            height: 29,
            src: "",
            srcSet: "",
            srcSetWebp: "",
            srcWebp: "",
            tracedSVG: "",
            width: 170
          }
        }
      }

    },
  };

  const component = renderer.create(<Blog {...hocProps} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
