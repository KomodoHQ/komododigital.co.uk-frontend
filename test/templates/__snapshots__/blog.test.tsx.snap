// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blog template is rendered 1`] = `
Array [
  <noscript>
    You have Javascript disabled. While it isn't needed, it will make your experience nicer.
  </noscript>,
  <div
    className="navigationBar"
  >
    <h1>
      <a
        aria-current="page"
        href="/"
        onClick={[Function]}
      >
        Komodo
      </a>
    </h1>
    <div
      className="hamburger"
    >
      <a
        aria-current="page"
        href="/"
        onClick={[Function]}
      >
        Menu
      </a>
    </div>
    <ul
      className="Menu"
    >
      <li>
        <a
          href="/about"
          onClick={[Function]}
        >
          About
        </a>
      </li>
      <li>
        <a
          href="/case-studies"
          onClick={[Function]}
        >
          Case Studies
        </a>
      </li>
      <li>
        <a
          href="/blog-list"
          onClick={[Function]}
        >
          Insights
        </a>
      </li>
      <li>
        <a
          href="/contact"
          onClick={[Function]}
        >
          Contact
        </a>
      </li>
    </ul>
  </div>,
  <div
    className="main"
  >
    <img
      width="100%"
    />
    <div
      className="cs-wrapper "
      style={
        Object {
          "backgroundColor": "inherit",
          "paddingBottom": "100px",
          "paddingTop": "100px",
        }
      }
    >
      <div
        className="Content-Section"
      >
        <h2>
          No Title Found
        </h2>
        <div>
          <div
            dangerouslySetInnerHTML={
              Object {
                "__html": "<p>Test</p>",
              }
            }
          />
        </div>
      </div>
    </div>
    <footer>
      <div
        className="komodo"
      >
        Komodo
      </div>
      <div
        className="address"
      >
        <span>
          Address
        </span>
        <address>
          Komodo Digital,
          <br />
          63 Westgate Rd,
          <br />
          Newcastle upon Tyne,
          <br />
          NE1 1SG
        </address>
        <address>
          Komodo | The Core
          <br />
          Science Central, Bath Ln,
          <br />
          Newcastle upon Tyne
          <br />
          NE4 5TF
        </address>
      </div>
      <div
        className="sitemap"
      >
        <span>
          Sitemap
        </span>
        <ul>
          <li>
            <a
              aria-current="page"
              href="/"
              onClick={[Function]}
            >
              Home
            </a>
          </li>
          <li>
            <a
              aria-current="page"
              href="/"
              onClick={[Function]}
            >
              Case Studies
            </a>
          </li>
          <li>
            <a
              aria-current="page"
              href="/"
              onClick={[Function]}
            >
              Insights
            </a>
          </li>
          <li>
            <a
              aria-current="page"
              href="/"
              onClick={[Function]}
            >
              Careers
            </a>
          </li>
        </ul>
        <ul>
          <li>
            <a
              aria-current="page"
              href="/"
              onClick={[Function]}
            >
              Contact
            </a>
          </li>
          <li>
            <a
              href="https://twitter.com/komodohq"
            >
              Twitter
            </a>
          </li>
          <li>
            <a
              href="https://facebook.com/komodohq"
            >
              Facebook
            </a>
          </li>
          <li>
            <a
              href="https://linkedin.com/komodohq"
            >
              LinkedIn
            </a>
          </li>
        </ul>
      </div>
    </footer>
    <footer
      className="legal-footer"
    >
      <div
        className="company"
      >
        <span>
          Komodo Digital Copyright Â© 2018
        </span>
        <br />
        <a
          aria-current="page"
          href="/"
          onClick={[Function]}
        >
          Privacy Policy
        </a>
         | 
        <a
          aria-current="page"
          href="/"
          onClick={[Function]}
        >
          Terms and Conditions
        </a>
      </div>
      <div
        className="contact"
      >
        Get in touch on:
        <br />
        <a
          href="/tel:0191 228 6555"
          onClick={[Function]}
        >
          0191 228 6555
        </a>
      </div>
    </footer>
  </div>,
  <noscript>
    <img
      alt=""
      height="1"
      src="https://dc.ads.linkedin.com/collect/?pid=377068&fmt=gif"
      width="1"
    />
  </noscript>,
]
`;
